common/types/endpoint.go

```diff
@@ -343,3 +414,38 @@ func (e *Endpoint) InvalidatePolicy() {
 		e.Consumable.Iteration = 0
 	}
 }
+
+func (e *Endpoint) LogStatus(code StatusCode, msg string) {
```

Comment:
Hmm that's true. But how can we define if an endpoint is functional? I was assuming that, since the last endpoint operation was OK then the endpoint is operational.

I consider the log status messages are relevant and sufficient, so far so good.

What I propose then is to develop function that performs particular tests (TBD) on that endpoint. That function/tests would be triggered whenever the user inspects the endpoint or lists the endpoints on the host. The result of those tests would then result on your proposal for status health (OK, Build, Warn, Fail)